### ISSUE TITLE
AdditionalImageStores configuration

### ISSUE THREAD

**Issue by meox** [containers/storage#1306]

**Comment by rhatdan** [comment]
I would be ok with this.  BUT storage.conf right now does not handle overriding fields.  We only replace and fall back to defaults.  This is different then what we do in containers.conf and registries.conf, I believe.

Changing this to inheritance my be a breaking change.

Changing it to only inherit from storage.conf.d/* Would not be a breaking change, though.

/usr/share/containers/storage.conf -> /etc/containers/storage.conf/ -> $HOME/.config/storage.conf override completely

/etc/containers/storage.conf.d/* and $HOME/.config/storage.conf/.d/* would only override fields in the default storage.conf.

**Comment by meox** [comment]
Overrides in which way?
I mean if I have multiple have multiple storage.conf.d files and then I run `podman images` I will able to see all the images from all storage?

**Comment by rhatdan** [comment]
Right now if I have /usr/containers/storage.conf and /etc/containers/storage.conf, None of the fields in /usr/containers/storage.conf are used or even read.
We would need to change this.  So that fields in /etc/containers/storage.conf/.d would be merged (replacing just the specified fields) that are present in /etc/containers/storage.conf.

If two drop in files used the same fields, then the last one read would win.

**Comment by meox** [comment]
I'd like to have the opportunity to add multiple storage (AdditionalImageStores) just adding multiple config files in a folder.
What @rhatdan said seems not able to cover my proposed use case.

**Comment by rhatdan** [comment]
Yes you would have to add multiple images to a single file.

**Comment by meox** [comment]
> Yes you would have to add multiple images to a single file.

ok but is just like now if I understand correctly

**Comment by rhatdan** [comment]
Best way to make this happen is to open a PR.

**Comment by jacksgt** [comment]
Alright, at least it's good to know that a PR is welcome for this :-)

**Comment by rhatdan** [comment]
Yes I think this is a good idea.  If we had it to do all over again, I think we would have done storage.conf the same as containers.conf.

Since .d directories are a new feature, I see them as enhancing the storage.conf file rather then replacing.
